src/Middleware/WebSockets/src/HandshakeHelpers.cs

```diff
@@ -114,17 +123,18 @@ public static string CreateResponseKey(string requestKey)
 
             using (var algorithm = SHA1.Create())
             {
-                string merged = requestKey + "258EAFA5-E914-47DA-95CA-C5AB0DC85B11";
-
                 // requestKey is already verified to be small (24 bytes) by 'IsRequestKeyValid()' and everything is 1:1 mapping to UTF8 bytes
                 // so this can be hardcoded to 60 bytes for the requestKey + static websocket string
                 Span<byte> mergedBytes = stackalloc byte[60];
-                Encoding.UTF8.GetBytes(merged, mergedBytes);
+                Encoding.UTF8.GetBytes(requestKey, mergedBytes);
+                _encodedWebSocketKey.CopyTo(mergedBytes.Slice(24));
 
                 Span<byte> hashedBytes = stackalloc byte[20];
                 var success = algorithm.TryComputeHash(mergedBytes, hashedBytes, out var written);
-                Debug.Assert(success);
-                Debug.Assert(written == 20);
+                if (!success || written != 20)
+                {
+                    throw new InvalidOperationException("Could not compute the hash for the 'Sec-WebSocket-Accept' header.");
```

Comment:
Is this the right behavior? a 500?

Reply:
Probably. We don't know why it would fail and would have to debug it when it did.

Reply:
This should be a 400 but we can wait until it happens 😄